### DISCUSSION TITLE
Performance benchmark of `bw6_767`

### DISCUSSION THREAD

**Discussion by alv-around** [Consensys/gnark-crypto#335] [General]
I am researching on the field of 2-Pairing SNARKs. It has come. 
From my understanding any `bls12` curve has a matching `bw6` curve for that allows for composing. 
On the [gnark docks](https://docs.gnark.consensys.net/Concepts/schemes_curves#bls12-377-and-bw6-761-curves), it is stated that the `bls12_381` can not be efficiently paired.
I was wondering how inefficient they are compared to efficient curves such as `bls12_377`/`bw6_761`.
I noticed that there were some developments on on the `bw6_767` on this [repo](https://github.com/yelhousni/gnark-crypto/tree/feat/bw6_on_bls12-381/ecc/bw6-767).
So I was wondering if you could provide some valuable insight on the topic. 
Specially, since `bls12_381` seems to be the most accepted pairing-curve atm (Eth2.0, Algorand, Tezos etc..), so any kind of benchmark would be very useful :)

Looking forward your valuable insight,
thanks in advance!

**Comment by yelhousni** [comment]
Hi @alvaro-alonso, sorry I haven't seen your question before. Indeed any BLS12 has a matching BW6, but also any BLS24 and BN as we generalised this construction in this [paper](https://eprint.iacr.org/2022/586.pdf) (see section 5.2 and appendix C). However, in order for the BW6 to be efficient for SNARKs, its subgroup order `r` should be highly 2-adic i.e. `2^L | r-1` for a big integer `L`. This means, in the context of 2-chains, that the BLS12 (or BLS24 or BN) should be defined over a field of size `p` (`=r`) highly 2-adic. This is the case for BLS12-377 as `2^46 | p-1` but not for BLS12-381 as only `2^1 | p -1`.

**Reply by alv-around** [reply]
@yelhousni thanks for the response :) I was aware of the highly 2-adic `r`, but your comparison of the value `p` gives me a better idea of the magnitudes. 🙏
